[Yap Shi Hao] Duke Increments

shihaoyap opened the pull request:
Duke, Level 1-6

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

shihaoyap committed:
Level-1

shihaoyap committed:
Level-2

shihaoyap committed:
Level-3

shihaoyap committed:
Level-4

shihaoyap committed:
Level-4(Updated)

shihaoyap committed:
Level-5

shihaoyap committed:
Level-6

shihaoyap committed:
Level-7

shihaoyap committed:
Level-8

shihaoyap committed:
Revert "Level-7"

This reverts commit 4d1c61698c27ff97ae2197409218802818ce5a43.

shihaoyap committed:
Merge branch 'branch-Level-8'

shihaoyap committed:
Merge branch 'branch-Level-8' into HEAD

# Conflicts:
#	src/main/java/Duke.java

shihaoyap committed:
Resolve Merge Conflict

shihaoyap committed:
Merged-Level-7,8

shihaoyap committed:
A-MoreOOP

shihaoyap committed:
A-JUnit

shihaoyap committed:
A-MoreOOP#2

shihaoyap committed:
A-Jar-New-Release

shihaoyap committed:
branch-A-JavaDoc

shihaoyap committed:
Revert "branch-A-JavaDoc"

This reverts commit 2367fa3b3fb3f9e93bd171224623d1740bd4a89d.

shihaoyap committed:
Revert "Revert "branch-A-JavaDoc""

This reverts commit a19d6a75664e90dfd6b211043fc8ad1147465743.

shihaoyap committed:
branch-A-JavaDoc-Commit1

shihaoyap committed:
Indentation Change

shihaoyap committed:
branch-A-CodingStandard-Commit1

shihaoyap committed:
branch-A-CodingStandard-Commit1

shihaoyap committed:
branch-A-CodingStandard-Commit2

shihaoyap committed:
Minor Edits

shihaoyap committed:
branch-Level-9-Commit1

shihaoyap committed:
Merge branch 'branch-Level-9'

shihaoyap committed:
Merge branch 'branch-A-CodingStandard'

shihaoyap committed:
Merge branch 'branch-A-JavaDoc'

shihaoyap commented:
Duke, Week 3
Completed: Level-7, Level-8, A-MoreOOP, A-Packages, A-JUnit, A-Jar, A-JavaDoc, A-CodingStandard, Level-9, Create a PR to the upstream repo

shihaoyap committed:
Merge remote-tracking branch 'origin/gradle'

shihaoyap committed:
Merge Gradle and Master

shihaoyap committed:
A-Gradle and A-CodingStandard

shihaoyap committed:
Week3-Task-Completed

shihaoyap committed:
A-CheckStyle

shihaoyap committed:
A-JavaFx

shihaoyap committed:
A-JavaFx-Updated

shihaoyap committed:
Level-10

shihaoyap commented:
Duke, Week 4
Completed: A-Gradle, A-CheckStyle, Level-10, A-JavaFx

shihaoyap committed:
Level-10-Updated

cheongliyuan commented:
Hi Shi Hao,

I did not notice any coding standard violations. However, trying _done 1_ caused an error. 
![Capture](https://user-images.githubusercontent.com/54430657/64075601-c4589a80-ccec-11e9-90fe-52353d4c7f13.PNG)
I believe it is due to the file location being specified according to how it was placed in your pc. 
![Capture2](https://user-images.githubusercontent.com/54430657/64075626-2913f500-cced-11e9-8510-bd15616d323b.PNG)
Running your program on other computers might fail if the folder is not setup accordingly. A simple solution would be to simply not state the directory, and it will be created where the main jar is(?). Should consider changing it, as it might not work for the tester! 

Moving on, i really liked the program! Having a GUI and being able to 'talk' to the robot gives it so much creativity points haha! 
Readability of code is good too, with different levels of abstractions being observed. Perhaps considering putting the _if (inputCommand.equals("bye"))_ in _getResponse()_ to another method? 

Looking forward to seeing your program develop. 
-Li Yuan

shihaoyap committed:
Updated-Code-According-To-Review

shihaoyap commented:
Thank you @cheongliyuan for your comments. The points that you have made are really helpful! I have make the necessary changes and updates to the code! Glad to hear that you have enjoyed the program! :)

shihaoyap committed:
Image-Update

CarbonGrid commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,59 @@
+    public void setDuke(Duke d) {
+        duke = d;
+        dialogContainer.getChildren().addAll(
+                DialogBox.getDukeDialog(duke.getStartMessage(), dukeImage));
```
Good effort for displaying the Start Message when duke is being initialized.

CarbonGrid commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,66 @@
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage("D:\\NUS Computer Science\\CS2103T\\duke\\src\\main\\java\\duke.txt");
```
Prefer relative path over full path. Full path are hardly usable in deployments.
```suggestion
        storage = new Storage("duke.txt");
```

CarbonGrid commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,44 @@
+public class Storage {
+
+    private File taskFile;
+    private String filePath;
```
Either taskFile or filePath should be used instead of "duke.txt" in Storage methods.

CarbonGrid reviewed (commented):
Good evening @shihaoyap ,

I think your coding standard is on the mark since checkStyle is not throwing any errors. Most expected functionalities are available but rather restrictive.
For example:
" done 1" doesn't work.
"done -1" sliently fails and doesn't clear userInput.
& Some GUI problems with restrictive parsing.
![Capture](https://user-images.githubusercontent.com/16153274/64105021-fa0e8980-cda7-11e9-9e02-39929a787e62.PNG)

Some key pointers:
1. Ui has too many repetitions of System.out.print("_____");
2. Ui is doing both Parser's and Storage's jobs.
Try to Let Ui work only on I/O. Storage on Files. and Parser on Parsing.
3. Command is preferably helpful to tackle Pointer 2 as per iP project guide.
4. Duke doesn't load or auto-load save data as per requirement in Level-7.
5. More JUnit tests could be implemented. e.g fixedTests() & runTest()
6. (It's just my preference) As per DukeException, when you have too many "if else" chained, opt for a switch case.

Anyways, good effort for the originality.
Keep up the effort and learning experience.

shihaoyap committed:
A-Gradle

shihaoyap committed:
Improved OOP and UI

shihaoyap committed:
Improved OOP codes

shihaoyap committed:
A-Assertions

shihaoyap committed:
A-CodeQuality

shihaoyap committed:
Week 5 : Updated, Merged

shihaoyap committed:
Week 5

shihaoyap committed:
A-CodeQuality (#5)

shihaoyap committed:
Merge commit '0a446fad28af8f3ac49a42b2dab29fd55ecebf48'

shihaoyap committed:
Week5

shihaoyap committed:
A-Travis

shihaoyap committed:
A-Travis-2

shihaoyap committed:
C-DetectDuplicates

shihaoyap committed:
C-BetterSearch

shihaoyap commented:
Duke, Week 4
Completed: Improved OOP Structure, Added Command Classes, Added Packages, A-Gradle, A-Assertions, A-CodeQuality, A-Travis, C-DetectDuplicates, C-BetterSearch

C-DetectDuplicates 
Program can detect if the user tries to input another event with the same description. Program will prompt user to use a unique description and will not allow him to use the same description.

 C-BetterSearch
Program can now search for all task that contains certain keywords, even though users only search partial descriptions.

shihaoyap committed:
Set theme jekyll-theme-midnight

shihaoyap committed:
Finalize the features -  Improve UI, Improve I/O Exception Handling

shihaoyap committed:
Merge branch 'master' of https://github.com/shihaoyap/duke

shihaoyap committed:
no message

shihaoyap committed:
no message

shihaoyap committed:
Ui.png and README.md

shihaoyap committed:
no message

shihaoyap committed:
Set theme jekyll-theme-midnight

shihaoyap committed:
no message

shihaoyap committed:
Merge branch 'master' of https://github.com/shihaoyap/duke

shihaoyap committed:
no message

shihaoyap committed:
no message

shihaoyap committed:
no message

shihaoyap committed:
no message

shihaoyap committed:
Week 6 Task - Ui.png, A-UserGuide, C-Sort

shihaoyap committed:
no message

shihaoyap committed:
A-Release

shihaoyap committed:
Update Travis-ci

shihaoyap committed:
Update to fix Travis-CI

shihaoyap committed:
Improved A-Sort to allow sort by task type and time. Changed image to non copy righted free images. Updated Ui.png and README.md

shihaoyap committed:
Cross-Platform fix

shihaoyap committed:
Update Build
